File: RssService/models/Record.py
from enum import Enum


class StateType(Enum):
    Unknown = -1
    Unread = 0
    UnreadModified = 1
    Readed = 2

class Record():
    def __init__(self, id, feed_id, link, date, title, description, state=StateType.Unknown):
        self.id = id
        self.feed_id = feed_id
        self.link = link
        self.date = date
        self.title = title
        self.description = description
        self.state = state

    def __init__(self, d):
        self.id = d["id"]
        self.feed_id = d["feed_id"]
        self.link = d["link"]
        self.date = d["date"]
        self.title = d["title"]
        self.description = d["description"]
        self.state = d["state"]

    def dict(self):
        result = {
            "id": self.id,
            "feed_id": self.feed_id,
            "link": self.link,
            "date": self.date,
            "title": self.title,
            "description": self.description,
            "state": self.state
            }
        return result;

File: RssCollector/record.cpp
#include "record.hpp"
#include "md5.hpp"
#include "str_utils.hpp"
#include <ctime>

namespace models
{
	record::record()
	{
	}

	record::record(const char* id,
			const char* link,
			const long date,
			const char* title,
			const char* description)
	{
			this->id = std::string(id);
			this->link = std::string(link);
			this->title = std::string(title);
			if (description == NULL)
			{
			    this->description = std::string("");
			}
			else
			{
			    this->description = std::string(description);
			}

			this->hash = md5(this->id);
			this->date = date;
	}

	std::string record::get_id() const
	{
			return id;
	}


	std::string record::get_title() const
	{
			return title;
	}
	std::string record::get_link() const
	{
			return link;
	}

	std::string record::get_description() const
	{
			return description;
	}

	long record::get_date() const
	{
			return date;
	}

	std::string record::get_hash() const
	{
			return hash;
	}

	std::string record::md5(const std::string &input)
	{
		return utils::md5(input);
	}

	/*std::wstring record::date_str()
	{
			std::wstring result;
			boost::local_time::local_time_facet* lf(new boost::local_time::local_time_facet("%a, %d %b %Y %H:%M:%S GMT"));
			try
			{
					std::wstringstream ss;
					ss.imbue(std::locale(ss.getloc(), lf));
					ss << m_dt;
					result = ss.str();
					std::wcout << result;
			}
			catch (std::exception& e)
			{
					std::cout << "Exception: " << e.what() << std::endl;
			}
			return result;
	}*/
}
